util/test/start_test.py

```diff
@@ -327,8 +327,12 @@ def test_directory(test, test_type):
                         are_tests = True
                         break
 
+            run_local_sub_test = os.access(os.path.join(dir, "sub_test"), os.X_OK)
+            if args.performance:
```

Comment:
I have been wondering if this should be `if test_type == "performance":` (since that occurs a bit earlier in the file).

Here is my understanding: when we do performance testing, we run two stages: `"performance"` and then `"graph"`. Vs, in regular correctness testing, we just one run stage, called `"run"`.

See also

https://github.com/chapel-lang/chapel/blob/cc321a176e335ac1b158dd7ccbf214e47dfa0955/util/test/start_test.py#L147-L158

I suspect that the custom `sub_test` should be run in neither of those, though. It also looks like, if we use `--gen-graphs`, we wouldn't want these `sub_test`s to run, right? In that case I think we are better off with `if test_type == "performance" or test_type == "graph":` (or writing the inverse -- `if test_type != "run":`)

Reply:
Good catch, thanks Michael!